### ISSUE TITLE
Support for changing existing application registrations

### ISSUE THREAD

**Issue by bt701** [microsoftgraph/msgraph-bicep-types#195]
**Is your feature request related to a problem? Please describe.**

Currently the permissions required for creating an application registration is quite big (ReadWrite.All), so is unlikely to be granted to individuals (as it implies you could edit other peoples registrations).

A workaround might be to have the 'skeleton' one provisioned, then be granted owner, and thus drive to a desired state once you 'own' the app registration.

Eg something like
```
resource application 'Microsoft.Graph/applications@v1.0' existing = {
  uniqueName: 'myappRegName'
  
  web: {    
    redirectUris: [ 'https://mycoolapp' ]
  }
}
```

Unfortunately, this doesn't work

**Describe the solution you'd like**

Support for 'existing' to manipulate the settings of the application type

Currently it returns an error
`Error BCP173: The property "web" cannot be used in an existing resource declaration. [https://aka.ms/bicep/core-diagnostics#BCP173]`

**Additional context**

Another solution might be a way for the bicep resources to have a some sort of 'CreateNew' permission, instead of ReadWriteAll. That way teams could be given 'CreateNew' and then only manipulate resources they own via ReadWriteOwner

**Comment by bt701** [comment]
Ah yes, I think I may be mis-using existing there, it was an attempt to try and update resources that already existed in Entra, I usually experience bicep as not liking if an existing named resource exists (eg unless defined in that bicep or referenced using existing), does this mean that if uniqueName is set (without existing) we will be updating existing registrations?. I have some questions around what that would mean for removing properties, eg would the graph bicep clear fields if they are not-set in the bicep?

For permissions, 
[Application.ReadWrite.All]https://learn.microsoft.com/en-us/graph/permissions-reference#applicationreadwriteall
This appears you can delete others app registrations, eg you can nuke everything, it's an admin role
[Application.ReadWrite.OwnedBy](https://learn.microsoft.com/en-us/graph/permissions-reference#applicationreadwriteall)
This is fantastic, you can modified applications you own, but if it doesn't exist. You seem to need the 'god' level permission
[Application.Create] -->Missing. Ideally this would be perfect, allowing users to create applications, and then manage them, but not other applications they are not an owner of.

**Comment by eketo-msft** [comment]
Yes, if you have uniqueName set then it will update the application registration. 

Graph Bicep won't clear the fields which are not defined.  We want to avoid breaking existing resources if they are not fully defined in the template.  It would be challenging to do this as adding properties to a resource in Graph happens often and is not considered a breaking change so there will be drift between the resource defined in the template and in the Graph metadata.

Re: permissions, we'll provide this feedback to the product owners to discuss further.

**Comment by dkershaw10** [comment]
Sorry for the late response @bt701.
Following up on the **permissions** questions.

Please clarify if you are intending to deploy the bicep file using a signed-in user or using an application (zero-touch deployment)?  
- If it's the former I'm a little puzzled by the question.
Currently, only Az PS and Az CLI can deploy Bicep files (containing Graph resources) with a signed-in user, which means you can't use your own app to deploy a Bicep file. Assigning delegated permission to an app that is used to interactively deploy resources would only become relevant once we fix the issue to allow your own app to perform a deployment (we hope the fix is in place CY2025Q1).
- If it's the latter (app-only), then Application.ReadWrite.OwnedBy should do the trick, as it should allow creation of new applications.  Are you saying that is not the case?
